src/components/app/data/hooks/useAlgoliaSearch.ts

```diff
@@ -189,30 +191,35 @@ const useAlgoliaSearch = (indexName: string | null = null): AlgoliaWithCatalogFi
     isIndexSupported,
   } = useSecuredAlgoliaMetadata(indexName);
 
-  const shouldUseSecuredAlgoliaApiKey = (
-    isCatalogQueryFiltersEnabled
-    && isIndexSupported
-    && !!securedAlgoliaMetadata.securedAlgoliaApiKey
-  );
-
-  // Based on the waffle flag and supported indexes, we will use the secured algolia
-  // key or default back to the legacy initialization of the search client and indexes
-  // The fallback once the secured algolia api key has been rolled out to all users via a waffle flag
-  // (catalogQuerySearchFiltersEnabled) is considered an errored state where downstream components would
-  // display the <SearchUnavailableAlert /> if no search client is returned or the upstream secured
-  // algolia api call fails from the BFF.
-  const algoliaSearchApiKey: string = shouldUseSecuredAlgoliaApiKey
-    ? securedAlgoliaMetadata.securedAlgoliaApiKey!
-    : config.ALGOLIA_SEARCH_API_KEY;
-
   // Update instantiate search client with or without a secured
   // algolia api key and retrieve the initialized algolia index
   return useMemo(() => {
+    const shouldUseSecuredAlgoliaApiKey = (
+      isCatalogQueryFiltersEnabled
+      && isIndexSupported
+      && !!securedAlgoliaMetadata.securedAlgoliaApiKey
+    );
+
+    const hasCatalogUuidToCatalogQueryUuidMapping = !isObjEmpty(
+      securedAlgoliaMetadata.catalogUuidsToCatalogQueryUuids,
+    );
```

Comment:
[curious] Is there any reason why this variable couldn't be part of the `shouldUseSecuredAlgoliaApiKey` itself? 

Alternatively, based on your earlier finding for the secured Algolia key API throwing an error when there are no catalogs (and thus no `securedAlgoliaMetadata.catalogUuidsToCatalogQueryUuids` and no `securedAlgoliaMetadata.securedAlgoliaApiKey`), could solely rely on the existing `shouldUseSecuredAlgoliaApiKey` instead? 

It feels like the check for `hasCatalogUuidToCatalogQueryUuidMapping` is essentially equivalent to the existing check `!!securedAlgoliaMetadata.securedAlgoliaApiKey` on its own, given the API wouldn't return a `securedAlgoliaApiKey` without catalogs anyways? 🤔